### PR TITLE
Cooks assistant

### PR DESCRIPTION
**Pull request by RealPoke**
Added the quest cooks assistant, this will complete cooks assistants without any requirements.

### CODE REVIEW

**Comment by RealPoke** [comment]
Do you want me to fix these two issues that codacy came up with?

1. There was no default statement in the switch in the example, I could make the default throw an exception or something.
2. I have never heard about "NPath complexity" but I can split things out into their own functions and such, I just find the commenting fine.

**Comment by Explv** [comment]
@RealPoke I will review this PR this evening, thanks for the contribution

File: AIO/src/org/aio/activities/quests/CooksAssistant.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+			// Put grain
+			if(!put && !operated && getInventory().contains("Grain")) {
+				
+				if(upper.contains(myPlayer())){
+					
+					if(getInventory().getSelectedItemName() != null && getInventory().getSelectedItemName().equals("Grain")) {
+						
+						RS2Object hopper = getObjects().closest(n -> n.getName().equals("Hopper"));
+							
+						if(hopper != null && hopper.interact("Use"))
```
Comment (Explv) [inline AIO/src/org/aio/activities/quests/CooksAssistant.java]: Make sure you put curly braces around all conditional statements `{}`

File: AIO/src/org/aio/activities/quests/CooksAssistant.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+		        getWalking().webWalk(Bank.getAreas());
+		    }
+		    
+		} else {
+        
+	        if (getTabs().getOpen() != Tab.INVENTORY) {
+	            getTabs().open(Tab.INVENTORY);
+	        }
+	        
+	        switch (getProgress()) {
```
Comment (Explv) [inline AIO/src/org/aio/activities/quests/CooksAssistant.java]: You can add a default branch to this switch statement:

```
default:
    log("Unrecognised progress config value: " + getProgress());
    setFailed();
    return;
```

File: AIO/src/org/aio/activities/quests/CooksAssistant.java [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        } else {
+        	completeDialog("Cook", cookOptions);
+        }
+    }
+
+    private void completeDialog(String npcName, String... options) throws InterruptedException {
+        if(!getDialogues().inDialogue()){
+            talkTo(npcName);
+        } else if(getDialogues().isPendingContinuation()){
+            getDialogues().clickContinue();
```
Comment (Explv) [inline AIO/src/org/aio/activities/quests/CooksAssistant.java]: You don't need this, `completeDialogue` will click continue

File: AIO/src/org/aio/activities/quests/CooksAssistant.java [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        NPC cook = getNpcs().closest("Cook");
+
+        if(cook == null){
+        	getWalking().webWalk(cookRoom);
+        } else {
+        	completeDialog("Cook", cookOptions);
+        }
+    }
+
+    private void completeDialog(String npcName, String... options) throws InterruptedException {
```
Comment (Explv) [inline AIO/src/org/aio/activities/quests/CooksAssistant.java]: Can you move this code into the `talkToCook()` function?

File: AIO/src/org/aio/activities/quests/CooksAssistant.java [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+				getItemFromObject(bin, "Pot of flour", "Flour bin", "Empty");
+				
+			}
+			
+		}
+    	
+    }
+    
+    private void getItemFromObject(Area place, String itemName, String objectName, String interaction) throws InterruptedException {
```
Comment (Explv) [inline AIO/src/org/aio/activities/quests/CooksAssistant.java]: This function and `getGroundItem` are very similar. You could combine these into one function by adding a boolean parameter `isGroundItem` or something along those lines

**Comment by Explv** [comment]
Logic was all there, just needed a bit of refactoring.

Thanks again for the contribution!

**Pull request merged**